video/mp_image.c

```diff
@@ -844,6 +844,17 @@ bool mp_image_params_equal(const struct mp_image_params *p1,
            mp_rect_equals(&p1->crop, &p2->crop);
 }
 
+bool mp_image_params_static_equal(const struct mp_image_params *p1,
+                                  const struct mp_image_params *p2)
+{
+    // Compare only static video parameters, excluding dynamic metadata.
+    struct mp_image_params a = *p1;
+    struct mp_image_params b = *p2;
+    a.repr.dovi = b.repr.dovi = NULL;
+    a.color.hdr = b.color.hdr = (struct pl_hdr_metadata){0};
+    return mp_image_params_equal(&a, &b);
```

Comment:
Not sure it makes sense to do it this way. Would it make more sense to have `mp_image_params_static_equal` compare all its relevant things, then have `mp_image_params_equal` call `mp_image_params_static_equal` and also compare `params.repr.dovi` and `params.color.hdr`?

Reply:
We use `pl_color_repr_equal` and `pl_color_space_equal`, so we don't have fine-grained control how those are compared. Hence the clearing of fields. I didn't want to duplicate them.